runtime/jsTest/src/kotlinx/serialization/DynamicSerializerTest.kt

```diff
@@ -0,0 +1,146 @@
+package kotlinx.serialization
+
+import kotlinx.serialization.internal.IntArraySerializer
+import kotlinx.serialization.internal.IntSerializer
+import kotlinx.serialization.json.Json
+import kotlin.test.*
+
+
+class DynamicSerializerTest {
+    @Serializable
+    data class Data(val a: Int)
+
+    @Serializable
+    open class DataWrapper(open val s: String, val d: Data?)
+
+    @Serializable
+    data class NestedList(val a: String, val list: List<Int>)
+
+    @Serializable
+    data class ListOfLists(val l: List<List<Data>>)
+
+    @Serializable
+    data class MapWrapper(val m: Map<String, Int>)
+
+    @Serializable
+    data class ComplexMapWrapper(val m: Map<String, Data>)
+
+    @Serializable
+    data class WithChar(val a: Char)
+
+    @Serializable
+    data class AllTypes(
+            val b: Byte,
+            val s: Short,
+            val i: Int,
+            val f: Float,
+            val d: Double,
+            val c: Char,
+            val B: Boolean,
+            val S: String
+    )
+
+    @Test
+    fun dynamicSimpleTest() {
+        val data = Data(42)
+        val serialized = DynamicObjectSerializer().serialize(data)
+        assertEquals(Json.stringify(data), JSON.stringify(serialized))
+        assertEquals(data.a, serialized.a)
+
+        val data2 = WithChar('c')
+        val serialized2 = DynamicObjectSerializer().serialize(data2)
+        assertEquals(Json.stringify(data2), JSON.stringify(serialized2), "should serialize characters as strings")
+        assertEquals(data2.a.toString(), serialized2.a) // FIXME why is toString needed here, both should be char?
```

Comment:
Probably, because natively chars are represented as ints in Kotlin/JS. You can leave this `.toString()`